### DISCUSSION TITLE
Setting up rabbitmq using IPv6

### DISCUSSION THREAD

**Discussion by ALIEZZ1984** [rabbitmq/rabbitmq-server#11552] [Questions]
Rabbitmq Version - 3.13.3
Erlang Version - 26.2.5
I'm trying to setup a rabbitmq cluster on windows server using IPv6. I got to a point where I'm stuck when reading the document, wasn't sure where i needed to setup the Inter-node communication protocol. Wasn't sure if i needed to pass these variable by creating a config file or just using windows variables, and also wasn't sure about the erl_inetrc, file as it doesn't exist on my machine. Can you please provide some insight on this. also a snip of listeners showing its listening on IPv6
![image](https://github.com/rabbitmq/rabbitmq-server/assets/169689982/da8ac911-2319-4892-af3c-c43f697cd7ca)
 
![image](https://github.com/rabbitmq/rabbitmq-server/assets/169689982/bf8f0b8d-65c2-4f82-b94e-da39024ff8be)

**Comment by lukebakken** [comment]
https://www.rabbitmq.com/docs/configure#customise-environment
https://www.rabbitmq.com/docs/networking#inter-node-communication-protocol

* Log in as the user that installed RabbitMQ
* Create the `%AppData%\RabbitMQ\rabbitmq-env-conf.bat` file with this content. **BE SURE** to use the correct path by replacing `ADMIN` with the correct user name:
    ```
    @echo off
    set SERVER_ADDITIONAL_ERL_ARGS=-kernel inetrc C:/Users/ADMIN/AppData/Roaming/RabbitMQ/erl_inetrc
    set CTL_ERL_ARGS=-kernel inetrc C:/Users/ADMIN/AppData/Roaming/RabbitMQ/erl_inetrc
    ```
* Create the `C:/Users/ADMIN/AppData/Roaming/RabbitMQ/erl_inetrc` file with `{inet6,true}.` as specified in our docs.

Let me know how this works. IPv6 is rarely seen, especially in a Windows environment.

**Reply by ALIEZZ1984** [reply]
@lukebakken  - In the C:/Users/ADMIN/AppData/Roaming/RabbitMQ, for some reason, i don't have that path in my windows instance. I don't have the rabbitMQ Folder in the above specified path,  i replaced the ADMIN with the user i installed rabbitmq, I only have this path C:\Program Files\RabbitMQ Server\rabbitmq_server-3.13.3 ,and include the following folder, show i just create the batch and place it in this folder?
![image](https://github.com/rabbitmq/rabbitmq-server/assets/169689982/453c5bfb-dc7e-4df8-9100-d3229fc6c323)

**Reply by lukebakken** [reply]
No, that is not the correct location. You probably have not yet started RabbitMQ. You can manually create the directory that I indicated.

**Reply by ALIEZZ1984** [reply]
@lukebakken  - thanks, I will try and will update the thread with my finding. thanks. Once i create the file as per above, Do i need to do anything further, or just try and join the cluster?

**Reply by ALIEZZ1984** [reply]
Hi @lukebakken  - When i use the command - rabbitmqctl join_cluster rabbit@hostname.

![image](https://github.com/rabbitmq/rabbitmq-server/assets/169689982/10b1d3a7-7d47-4402-8a60-a7e07b1f1747)

also when i do the rabbitmq-diagnostics resolve_hostname ash-genalab304, it resolve backloop ipv4, although on my system i don't have IPv4 Interface enabled.

![image](https://github.com/rabbitmq/rabbitmq-server/assets/169689982/5f0ca7cf-167f-4c69-b3e9-c2b222cca4b0)

**Reply by lukebakken** [reply]
`nxdomain` means that the hostname `ash-genalab304` can't be resolved by the `epmd` process.

Please capture the output of these commands into a file, and _ATTACH_ the file to your response. Screenshots are not as useful:

```
ping ash-genalab304
ipconfig /all
```

**Reply by ALIEZZ1984** [reply]
@lukebakken  - Attached requested information.
[ash_genalab304_ipconfig_result.txt](https://github.com/user-attachments/files/15991074/ash_genalab304_ipconfig_result.txt)
[ash-genalab304_ping_result.txt](https://github.com/user-attachments/files/15991075/ash-genalab304_ping_result.txt)

**Reply by lukebakken** [reply]
Thanks for that information.

* Are you running Windows in a VM of some sort?
* If you go to the `Ethernet0` network adapter properties, I'm assuming that you _ONLY_ have "Internet Protocol Version 6" in the list. For reference, this is the dialog I'm talking about, and you get to it via Control Panel -> Network and Sharing Center -> Change Adapter Settings -> Right-click on adapter and choose "Properties"
![ethernet-properties](https://github.com/rabbitmq/rabbitmq-server/assets/514926/d8d29b79-2ff5-41af-affe-420d183fde3f)
----

Finally, this is the sort of issue / question that would be supported for a RabbitMQ user with a [commercial support contract](https://www.rabbitmq.com/contact#paid-support). I may be able to find time to set up an IPv6-only Windows environment, but it is literally the lowest priority item on Team RabbitMQ's very long list of priorities right now.

If I were you, I would start with these internet searches and pages to figure this issue out for yourself:
* https://www.google.com/search?q=erlang+epmd+ipv6+only
* https://groups.google.com/g/rabbitmq-users/c/9ir1WK-nYBA
* https://www.rabbitmq.com/docs/networking#epmd

In fact, I think the last link is what you need to do in your environment. You should change your `rabbitmq-env-conf.bat` file to the following:

```
@echo off
set ERL_EPMD_ADDRESS=::1
set SERVER_ADDITIONAL_ERL_ARGS=-kernel inetrc C:/Users/ADMIN/AppData/Roaming/RabbitMQ/erl_inetrc
set CTL_ERL_ARGS=-kernel inetrc C:/Users/ADMIN/AppData/Roaming/RabbitMQ/erl_inetrc
```

You should also set a *system* environment variable with the same value.

Then, ensure you have debug logging enabled. Create the `C:/Users/ADMIN/AppData/Roaming/RabbitMQ/rabbitmq.conf` file if you haven't already, with this content:

```
log.console = true
log.console.level = debug
```

Then, ensure that all `erl.exe`, `werl.exe` and `epmd.exe` processes are killed.

Then, run these commands and send me all of the output:

```
cd "C:\Program Files\RabbitMQ Server\rabbitmq_server-3.13.3\sbin"
set
set LOG=debug
.\rabbitmq-server.bat
```

**Reply by ALIEZZ1984** [reply]
Do i need to setup System environment variable. for these 3 below.
set ERL_EPMD_ADDRESS=::1
set SERVER_ADDITIONAL_ERL_ARGS=-kernel inetrc C:/Users/ADMIN/AppData/Roaming/RabbitMQ/erl_inetrc
set CTL_ERL_ARGS=-kernel inetrc C:/Users/ADMIN/AppData/Roaming/RabbitMQ/erl_inetrc

**Reply by lukebakken** [reply]
You can if you'd like, I just think that `ERL_EPMD_ADDRESS=::1` is the important one.

**Reply by ALIEZZ1984** [reply]
Ok. I will try those steps and get back to you with findings.
